### ISSUE TITLE
API: total number of results is not exposed

### ISSUE THREAD

**Issue by avivace** [indico/indico#5106]
We are trying to build a paginated call (using `limit` and `offset`) to the Indico API on the **/export/event** endpoint, but we can't find a way to know how many results a query yields without first sending at least one without the `limit` parameters, because "count" in the response always reports the number of results in the paginated query..

E.g.:
`GET https://indico.cern.ch/export/event/search/parisi.json?&limit=20` has "count=20" in the response, so I need to `GET https://indico.cern.ch/export/event/search/parisi.json` to get the total count

**Comment by ThiefMaster** [comment]
We can't give you a total result count here since that would require performing an access check for every event matching the search string, not just those that are returned - which would of course kill performance depending on the number of events matches, so I suggest you to use a high `limit` and/or keep fetching additional pages.

**Comment by avivace** [comment]
> We can't give you a total result count here since that would require performing an access check for every event matching the search string, not just those that are returned - which would of course kill performance depending on the number of events matches, so I suggest you to use a high `limit` and/or keep fetching additional pages.

How is this feasible when a query has a lot of results? `https://indico.cern.ch/export/event/search/particle.json` has an hard coded limit of 1000 results, so I have always to at least perform a get request, waiting and processing a JSON response payload of (at least) 200 KB to undertand I won't be able to paginate in a reliable way

**Comment by ThiefMaster** [comment]
You are right, paginating such results is not really feasible if you want indexed navigation. I would recommend to do a "next page" style navigation instead of numbered pages.

Also, what exactly is your use case here? Maybe using the (undocumented) [API](https://indico.cern.ch/search/api/search?q=particle&type=event) used by the [search feature](https://indico.cern.ch/search/) would be better? This one *does* have the kind of pagination you are looking for.

Maybe we should have a chat on Mattermost or Zoom to discuss your usecase...?

**Comment by avivace** [comment]
> You are right, paginating such results is not really feasible if you want indexed navigation. I would recommend to do a "next page" style navigation instead of numbered pages.
> 
> Also, what exactly is your use case here? Maybe using the (undocumented) [API](https://indico.cern.ch/search/api/search?q=particle&type=event) used by the [search feature](https://indico.cern.ch/search/) would be better? This one _does_ have the kind of pagination you are looking for.
> 
> Maybe we should have a chat on Mattermost or Zoom to discuss your usecase...?

That would be great! I can drop by your office if you have a minute
